examples/routing_service/routing_service_msql_adapter/cpp/ConnectorMsqlDb.cxx

```diff
@@ -0,0 +1,337 @@
+    retcode = SQLAllocHandle(SQL_HANDLE_ENV, SQL_NULL_HANDLE, &sqlEnvHandle);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Allocating the environment handle failed");
+        disconnect();
+        return connected_;
+    }
+
+    /* Set the ODBC version environment attribute */
+    retcode = SQLSetEnvAttr(
+            sqlEnvHandle,
+            SQL_ATTR_ODBC_VERSION,
+            (SQLPOINTER) SQL_OV_ODBC3,
+            0);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR(
+                "SQL Error - Setting the ODBC version environment attribute");
+        disconnect();
+        return connected_;
+    }
+
+    /* Allocates the connection handle */
+    retcode = SQLAllocHandle(SQL_HANDLE_DBC, sqlEnvHandle, &sqlConnHandle);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Allocating SQL database connection failed");
+        disconnect();
+        return connected_;
+    }
+
+    /* Sets login timeout to 5 seconds */
+    retcode = SQLSetConnectAttr(
+            sqlConnHandle,
+            SQL_LOGIN_TIMEOUT,
+            (SQLPOINTER) TIMEOUT,
+            0);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Setting connection timeout failed");
+        disconnect();
+        return connected_;
+    }
+
+    SQLCHAR retconstring[SQL_BUFF_LEN];
+    /* connect to SQL Server */
+    retcode = SQLDriverConnect(
+            sqlConnHandle,
+            NULL,
+            (SQLCHAR *) connectionInfo.data(),
+            SQL_NTS,
+            retconstring,
+            SQL_BUFF_LEN,
+            NULL,
+            SQL_DRIVER_NOPROMPT);
+
+    if (retcode == SQL_ERROR) {
+        msqlShowError(SQL_HANDLE_DBC, sqlConnHandle);
+        disconnect();
+        return connected_;
+    } else {
+        RTI_RS_LOG(connectionInfo.data());
+        RTI_RS_LOG("SQL connection to database successful");
+        connected_ = true;
+    }
+
+    retcode = SQLAllocHandle(SQL_HANDLE_STMT, sqlConnHandle, &sqlStmtHandle);
+
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Allocating the statement handle failed");
+        disconnect();
+        return connected_;
+    }
+    return connected_;
+}
+
+/*
+ * Free up resources and set connected_ to false
+ */
+void ConnectorMsqlDb::disconnect()
+{
+    RTI_RS_LOG_FN(disconnect_db);
+
+    SQLFreeHandle(SQL_HANDLE_STMT, sqlStmtHandle);
+    SQLDisconnect(sqlConnHandle);
+    SQLFreeHandle(SQL_HANDLE_DBC, sqlConnHandle);
+    SQLFreeHandle(SQL_HANDLE_ENV, sqlEnvHandle);
+
+    connected_ = false;
+}
+
+/*
+ * Set connection information and form connection string
+ */
+void ConnectorMsqlDb::setConnectionInfo(const PropertySet &properties)
+{
+    RTI_RS_LOG_FN(setConnectionInfo);
+
+    auto it = properties.find("msql.db.servername");
```

Comment:
our pattern for error handling is typically the opposite. We check for the wrong value first, log, and then return/throw. 

I understand that would be big changes for you. It's not required you them now if you don;t have time, but keep it in mind for the future

Reply:
This log statement is for tracing. If logging is on it will print that the method was called. It is not there to indicate an error happened. Most of the example has these tracing log methods to help the user.

Reply:
The only log method for errors is RTI_RS_ERROR()